tests/unit/shutdown.tcl

```diff
@@ -0,0 +1,49 @@
+start_server {tags {"shutdown"}} {
+    test {Temp rdb will be deleted if we use bg_unlink when shutdown} {
+        for {set i 0} {$i < 20} {incr i} {
+            r set $i $i
+        }
+        # It will cost 2s(20 * 100ms) to dump rdb
+        r config set rdb-key-save-delay 100000
+
+        # Child is dumping rdb
+        r bgsave
+        after 100
```

Comment:
Oh, is this problem?  yesterday, you pointed out my some similar problems, `after 100` freezes tcl code? 
https://github.com/redis/redis/pull/7819#discussion_r492080951

Reply:
the assertion that fails is the one in line 42 (first assert in the second test).
and in this case what could have happened is that the `after 100` was not enough, and we tested the existence of the rdb file before redis started to save it (since redis was for some reason frozen for some 100ms).

i guess we should change this to a wait_for_condition loop that will usually exit a lot sooner, but at the same time have longer timeout.
i suppose similar changes can be made in a few other places in these new tests (to reflect the spirit of the changes i asked for in #7819)

please make sure you can reproduce this (possibly bu pushing an experimental branch into github to use the actions) before attempting to solve it.